library/Fission/AWS/Route53.hs

```diff
@@ -0,0 +1,58 @@
+module Fission.AWS.Route53
+  ( createChangeRequest
+  , registerDomain
+  ) where
+
+import RIO
+
+import Control.Lens ((?~))
+import Data.Has
+import Servant
+
+import Network.AWS
+import Network.AWS.Auth as AWS
+import Network.AWS.Prelude
+import Network.AWS.Route53
+
+import qualified Fission.Config    as Config
+import           Fission.AWS.Types as AWS
+import           Fission.AWS
+import           Fission.Internal.Constraint
+
+registerDomain :: MonadRIO           cfg m
+               => MonadUnliftIO          m
+               => HasLogFunc         cfg
+               => Has AWS.AccessKey  cfg
+               => Has AWS.SecretKey  cfg
+               => Has AWS.ZoneId     cfg
+               => RecordType
+               -> Text
+               -> Text
+               -> m (Either ServerError ChangeResourceRecordSetsResponse)
+registerDomain recordType domain content = do
+  logDebug $ "Updating DNS record at: " <> displayShow domain
+  env <- createEnv
+  req <- createChangeRequest recordType domain content
+
+  withAWS env NorthVirginia $ do
+    res <- send req
+    return $ validate res
```

Comment:
Something in me wants to make this shorter

```haskell
 withAWS env NorthVirginia . pure $ validate =<< send req
```

But the current version reads more clearly than that, so 👍